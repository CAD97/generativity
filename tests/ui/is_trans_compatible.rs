// see cad97/generativity#13 and rust-lang/rust#78586
// if this fails in crater *please* ping me; I *want* this to not hit the lint!
#![deny(repr_transparent_external_private_fields)] // this should not trigger

use generativity::{make_guard, Id};

#[repr(transparent)]
pub struct BOption<'id, T>(Option<T>, Id<'id>); // this should work

fn main() {
    make_guard!(a);
    let _ = BOption(Some(0), a.into());
}
