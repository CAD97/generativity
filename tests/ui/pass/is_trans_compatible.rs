// see cad97/generativity#13
#![deny(repr_transparent_external_private_fields)]

use generativity::Id;

pub struct BOption<'id, T>(Option<T>, Id<'id>);

fn main() {}
