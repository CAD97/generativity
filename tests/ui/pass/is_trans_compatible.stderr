error[E0601]: `main` function not found in crate `$CRATE`
 --> tests/ui/is_trans_compatible.rs:5:48
  |
5 | pub struct BOption<'id, T>(Option<T>, Id<'id>);
  |                                                ^ consider adding a `main` function to `$DIR/tests/ui/is_trans_compatible.rs`
